### DISCUSSION TITLE
Summarisation Model loading

### DISCUSSION THREAD

**Discussion by YuryAlsheuski** [deepjavalibrary/djl#3168] [Q&A]
I would like to load facebook/bart-large-cnn from HuggingFace to create text summary.
I have two cases:
1. I zipped it via model_zoo_importer.py but have the next exception:
`Unsupported model architecture: BartForConditionalGeneration for facebook/bart-large-cnn.`
So my first question is - which architectures are supported by model_zoo_importer.py?
2. I successfully converted it to *.pt file, load it with djl without any exceptions, but how to understand which existed Translator I can use for text summarisation or how I create personal Translator to work with the dedicated LLM? I know I should to implement Translator interface but how to parse LLM data inside?
Thanks in advance for any ideas!

**Comment by frankfliu** [comment]
`model_zoo_importer.py` doesn't support text-generation task, see supported tasks: https://github.com/deepjavalibrary/djl/blob/master/extensions/tokenizers/src/main/python/model_zoo_importer.py#L26-L31

You need to manually trace the model to pytorch or convert to onnx.

You can use this as example to build your own LLM application: https://github.com/deepjavalibrary/djl/blob/master/examples/src/main/java/ai/djl/examples/inference/nlp/TextGeneration.java
